generate/generate_derived_functions.sh

```diff
@@ -1,11 +1,28 @@
 #!/usr/bin/env bash
 
+THIS_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )"
+cd $THIS_DIR
+
+if [[ "$OSTYPE" == "darwin"* ]]; then
```

Comment:
Just an idea: this is a switch.  It is disguised as an `if/else` block.  Wouldn't it be showing intent more clearly with

```bash
function assert_gsed_available() {
   if ! command -v gsed &> /dev/null; then
        echo "gsed could not be found (gsed is the GNU implementation of sed, required by this script, under MacOS)
        Please install it with:
            brew install gnu-sed
        "
        exit 1
    fi
}


case "$OSTYPE" in
  darwin)
      assert_gsed_available || exit 1
      SED_CMD=gsed
      ;;
  *) SED_CMD=sed;;
esac

Reply:
Good idea, I did this